### PR TITLE
Update google analytic

### PR DESCRIPTION
**Pull request by glmanhtu**

### CODE REVIEW

File: core/common/library/Google/Analytic.php [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+ * @since   1.0.0
+ * @license http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt
+ */
+namespace Phanbook\Google;
+use Phanbook\Models\Settings;
+
+/**
+*
+*/
+class Analytic extends \Phalcon\DI\Injectable
```
Comment (ghost) [inline core/common/library/Google/Analytic.php]: +class Analytic extends \Phalcon\DI\Injectable to class Analytic extends Injectable, then add use \Phalcon\DI\Injectable the above

File: core/common/library/Google/Analytic.php [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+{
+	private $client;
+	private $clientId;
+	private $clientSecret;
+
+	function __construct()
```
Comment (ghost) [inline core/common/library/Google/Analytic.php]: function __construct() should replace to public function __construct()

File: core/common/library/Google/Analytic.php [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        } catch (\Exception $e) {
+            return ["state" =>  false, "message"    =>  "Exception when get profile : ".print_r($e)];
+        }
+    }
+
+    public function getAnalyticData($accountID, $profileID)
+    {
+        $service = new \Google_Service_Analytics($this->client);
+
+        $from = date('Y-m-d', time()-30*24*60*60);; // 30 days
```
Comment (ghost) [inline core/common/library/Google/Analytic.php]: You have duplicate symbol ";"

File: core/common/library/Google/Analytic.php [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+        }
+        if ($this->client->isAccessTokenExpired()) {
+            if ($this->refreshToken()) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+        return true;
+    }
```
Comment (sergeyklay) [inline core/common/library/Google/Analytic.php:90]:
``` php
public function checkAccessToken()
{
    if (Settings::getAccessToken()) {
        if ($this->client->isAccessTokenExpired()) {
            return $this->refreshToken();
        }

        return true;
    }

    return  false;
}
```

-1 `if`
-1 `return`

:)

Comment (ghost) [inline core/common/library/Google/Analytic.php:90]: :+1:

File: core/common/library/Utils/Phanbook.php [hunk 5]
```diff
@@ -123,7 +125,7 @@ public function saveTagsInPosts($_tag, $object)
     /**
      * Retrieve a page give its title
      *
-     * @param  string       $page_title Page title
+     * @param  string $page_title Page title
```
Comment (sergeyklay) [inline core/common/library/Utils/Phanbook.php:128]: <s>$page_title</s> <b>$title</b>

File: core/common/models/Settings.php [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+
+    /**
+     * Independent Column Mapping.
+     * Keys are the real names in the table and the values their names in the application
+     *
+     * @return array
+     */
+    public function columnMap()
+    {
+        return array(
```
Comment (sergeyklay) [inline core/common/models/Settings.php:181]: `return [...];` We are in the 21st century :)

File: core/modules/backend/controllers/DashboardController.php [hunk 7]
```diff
@@ -30,6 +32,14 @@ class DashboardController extends ControllerBase
      */
     public function indexAction()
     {
+        $analytic = new Analytic();
+        // We check if user authorization
+        if ($analytic->checkAccessToken()) {
+            $this->view->isLogged = true;
+        } else {
+            $this->view->isLogged = false;
+        }
```
Comment (sergeyklay) [inline core/modules/backend/controllers/DashboardController.php:46]: `$this->view->isLogged = $analytic->checkAccessToken()` without needs of any `if`

File: core/common/models/Settings.php (hunk 6 above)
Comment (duythien) [inline core/common/models/Settings.php:181]: It created default by phalcon tool, you can fix this in phalcon tool to use late

File: core/common/library/Utils/Phanbook.php (hunk 5 above)
Comment (sergeyklay) [inline core/common/library/Utils/Phanbook.php:128]: @duythien  merged ?

Comment (ghost) [inline core/common/library/Utils/Phanbook.php:128]: I was forget it

**Pull request merged**